apps/script/HoC2023AiGenerateWeights.py

```diff
@@ -101,61 +112,71 @@
-
-background_output['pythonNames'] = background_output['output']
-palette_output['pythonNames'] = palette_output['output']
-foreground_output['pythonNames'] = foreground_output['output']
-
-# Modify output to reflect blockly id names.
-background_output['output'] = [background_effects_map[bg] for bg in background_output['output']]
-foreground_output['output'] = [foreground_effects_map[fg] for fg in foreground_output['output']]
-palette_output['output'] = [palettes_map[pal] for pal in palette_output['output']]
-
-with open("apps/static/dance/ai/model/cached-spacy-palette-map.json", "w") as json_file:
-    json_file.write(json.dumps(palette_output))
-
-with open("apps/static/dance/ai/model/cached-spacy-background-map.json", "w") as json_file:
-    json_file.write(json.dumps(background_output))
-
-with open("apps/static/dance/ai/model/cached-spacy-foreground-map.json", "w") as json_file:
-    json_file.write(json.dumps(foreground_output))
+
+# Create maps of model_descriptive_name to blockly_id and print lists
+# as 'model_descriptive_name | blockly_id | blockly_user_facing_name'
+# Updates lists in comments at bottom of this file.
+create_map_print_options(PALETTES, model_descriptive_names_map[PALETTES], palette_blockly_ids_list, palette_user_facing_names_list),
+create_map_print_options(BACKGROUND_EFFECTS, model_descriptive_names_map[BACKGROUND_EFFECTS], background_effect_blockly_ids_list, background_effect_user_facing_names_list),
+create_map_print_options(FOREGROUND_EFFECTS, model_descriptive_names_map[FOREGROUND_EFFECTS], foreground_effect_blockly_ids_list, foreground_effect_user_facing_names_list)
+
+# We use cache files to store ada's raw embedding outputs to reduce Open AI query costs.
+# An embedding is a vector (list) of floating point numbers, and embeddings are used to
+# measure the relatedness of text strings. The distance between two vectors measures
+# their relatedness. https://platform.openai.com/docs/guides/embeddings
+# These caches are stored as pickle files; python's native way to serialize data.
+embeddings_paths = {
+    EMOJIS: 'apps/static/dance/ai/model/input_embeddings.pkl',
+    PALETTES: 'apps/static/dance/ai/model/palette_embeddings.pkl',
+    BACKGROUND_EFFECTS: 'apps/static/dance/ai/model/background_embeddings.pkl',
+    FOREGROUND_EFFECTS: 'apps/static/dance/ai/model/foreground_embeddings.pkl'
+}
+
+embeddings_caches = {}
+for path in embeddings_paths.keys():
+    embeddings_caches[path] = load_embeddings_cache(path)
+
+# Retrieve embeddings using model_descriptive_names for input emojis, palettes, background effects, and foreground effects.
+options_lists = {
+    EMOJIS: emoji_model_descriptive_names_list,
+    PALETTES: model_descriptive_names_map[PALETTES],
+    BACKGROUND_EFFECTS: model_descriptive_names_map[BACKGROUND_EFFECTS],
+    FOREGROUND_EFFECTS: model_descriptive_names_map[FOREGROUND_EFFECTS]
+}
+
+embeddings = {}
```

Comment:
Nit: move this to just above the for loop that populates it (ie, have comment above this line)